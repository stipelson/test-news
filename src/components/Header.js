import { Link } from 'gatsby';
import PropTypes from 'prop-types';
import React from 'react';
import Navbar from 'emerald-ui/lib/Navbar';
import Nav from 'emerald-ui/lib/Nav';
import DropdownItem from 'emerald-ui/lib/DropdownItem';
import DropdownButton from 'emerald-ui/lib/DropdownButton';
import Avatar from 'emerald-ui/lib/Avatar';
// import { getParams } from '../lib/utils';

export const sections = [
  {
    topicId: '5dfccaa7-e8ab-4044-8355-b6bebba95499',
    name: 'Health',
  },
  {
    topicId: '240f6a12-b9d8-40a6-b1c6-a220e31d08de',
    name: 'Renewable Energy',
  },
  {
    topicId: 'b220679c-95ff-4e4e-a1fa-ad8b3905b7df',
    name: 'Cyber Security',
  },
  {
    topicId: 'c4277484-5aca-4db8-8c30-72ca9aac3ed3',
    name: 'Technology',
  },
];

const Header = ({
  siteTitle,
  brandLogo,
  params,
  onNavigate,
  menuNav,
  userNav,
}) => {
  const handleOnClick = (category) => {
    onNavigate(category);
  };

  let categoryParam = params && params.category ? params.category : '';

  const renderMenuNav = () => {
    return (
      <Nav grow collapsible>
        <DropdownButton title="Sections" id="dd1">
          {sections.map((section, index) => (
            <DropdownItem
              key={`section-dropdown-${index}`}
              eventKey="2"
              onClick={() => handleOnClick(section.topicId)}
              active={categoryParam === section.topicId}
              data-testid="section-action"
            >
              {section.name}
            </DropdownItem>
          ))}
        </DropdownButton>
        <Link to="/">Editorial</Link>
        <Link to="/">Contact us</Link>
      </Nav>
    );
  };

  const renderUserNav = () => {
    return (
      <Nav>
        <DropdownButton
          noCaret
          fromRight
          id="dd2"
          title={<Avatar title="JS" />}
        >
          <DropdownItem eventKey="1">Action</DropdownItem>
          <DropdownItem eventKey="2">Another action</DropdownItem>
          <DropdownItem eventKey="3" active>
            Active Item
          </DropdownItem>
          <DropdownItem separator />
          <DropdownItem eventKey="4">Separated link</DropdownItem>
        </DropdownButton>
      </Nav>
    );
  };

  return (
    <header
      style={{
        background: 'rebeccapurple',
        marginBottom: '1.45rem',
      }}
    >
      <Navbar fixedAtTop breakAt="sm">
        <div className="container display-flex">
          <Navbar.Brand>
            <Link to="/">
              <img src={brandLogo} alt={`Logo ${siteTitle}`} />
            </Link>
          </Navbar.Brand>
          {menuNav && renderMenuNav()}
          {userNav && renderUserNav()}
        </div>
      </Navbar>
    </header>
  );
};

Header.propTypes = {
  siteTitle: PropTypes.string,
  brandLogo: PropTypes.any,
  location: PropTypes.any,
  params: PropTypes.object,
  onNavigate: PropTypes.func,
  menuNav: PropTypes.bool,
  userNav: PropTypes.bool,
};

Header.defaultProps = {
  siteTitle: '',
  menuNav: false,
  userNav: false,
};

export default Header;
