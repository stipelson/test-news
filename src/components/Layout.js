/**
 * Layout component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useStaticQuery, graphql } from 'gatsby';

import BrandLogo from '../assets/images/logo.svg';
import Header from './Header';
import Alert from 'emerald-ui/lib/Alert';

const Layout = ({ children, hiddenTitle, params, onNavigate }) => {
  const [showAlert, setShowAlert] = useState(true);

  const { site } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
          }
        }
      }
    `
  );

  const alertMsg =
    'Welcome to the new look of News.com. Keep scrolling to discover interesting new features and news.';

  return (
    <>
      {hiddenTitle && (
        <h1 className="title-hidden">{site.siteMetadata.title}</h1>
      )}
      <Header
        siteTitle={site.siteMetadata.title}
        brandLogo={BrandLogo}
        params={params}
        onNavigate={onNavigate}
        menuNav
        userNav
      />
      {showAlert && (
        <div className="container">
          <Alert dismissible onDismiss={() => setShowAlert(false)}>
            <div>{alertMsg}</div>
          </Alert>
        </div>
      )}
      <main>{children}</main>
    </>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  hiddenTitle: PropTypes.bool,
  params: PropTypes.object,
  onNavigate: PropTypes.func,
};

Layout.defaultProps = {
  hiddenTitle: false,
};

export default Layout;
