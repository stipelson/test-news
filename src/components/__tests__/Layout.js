import React from 'react';
import { create, act } from 'react-test-renderer';
import { PureLayout as Layout } from '../Layout';
import Header from '../Header';
import Alert from 'emerald-ui/lib/Alert';

describe('Index render', () => {
  let component;
  let data;

  beforeAll(() => {
    data = {
      site: {
        siteMetadata: {
          title: 'Default Starter',
          description: 'Site description',
          author: 'site author',
        },
      },
    };
    component = create(
      <Layout data={data} menuNav={false} userNav={false}>
        <div />
      </Layout>
    );
  });

  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('render with header h1 hidden coorectly', () => {
    act(() => {
      component.update(
        <Layout data={data} menuNav={false} userNav={false} hiddenTitle>
          <div />
        </Layout>
      );
    });

    const instance = component.root;
    const header = instance.findByType('h1');

    expect(header.children).toEqual(['Default Starter']);
  });

  it('render without query data', () => {
    act(() => {
      component.update(
        <Layout menuNav={false} userNav={false} hiddenTitle>
          <div />
        </Layout>
      );
    });

    const instance = component.root;
    const title = instance.findByType('h1');
    const header = instance.findByType(Header);

    expect(title.children).toEqual(['Site title']);

    expect(header.props.siteTitle).toEqual('Site title');
  });

  it('render alert', () => {
    const instance = component.root;
    const alert = instance.findByType(Alert);
    const alertContent = alert.findByProps({ className: 'caption' });

    expect(alertContent.children).toEqual([
      'Welcome to the new look of News.com. Keep scrolling to discover interesting new features and news.',
    ]);
  });

  it('Close alert', () => {
    const instance = component.root;
    const alert = instance.findByType(Alert);

    act(() => {
      alert.props.onDismiss();
    });

    const alerts = instance.findAllByType(Alert);
    expect(alerts.length).toEqual(0);
  });
});
