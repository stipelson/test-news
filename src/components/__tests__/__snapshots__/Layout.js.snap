// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index render renders correctly 1`] = `
Array [
  <header
    style={
      Object {
        "background": "rebeccapurple",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      className="eui-navbar eui-navbar-light break-at-sm fixed-at-top"
      onClick={[Function]}
      role="presentation"
    >
      <div
        className="eui-navbar-bar"
      >
        <div
          className="container display-flex"
        >
          <div
            className="eui-navbar-brand"
          >
            <a
              href="/"
            >
              <img
                alt="Logo Default Starter"
                src="test-file-stub"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        className="eui-navbar-collapsed"
      />
    </div>
  </header>,
  <div
    className="container"
  >
    <div
      className="eui-alert eui-alert-info"
      role="alert"
    >
      <div
        className="caption"
      >
        Welcome to the new look of News.com. Keep scrolling to discover interesting new features and news.
      </div>
      <button
        className="eui-icon-button"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="eui-icon material-icons"
        >
          cancel
        </i>
      </button>
    </div>
  </div>,
  <main>
    <div />
  </main>,
]
`;
